dask/dataframe/shuffle.py

```diff
@@ -156,9 +156,9 @@ def barrier(args):
 def _set_partition(df, index, divisions, p, drop=True):
     """ Shard partition and dump into partd """
     df = df.set_index(index, drop=drop)
-    df = strip_categories(df)
     divisions = list(divisions)
     shards = shard_df_on_index(df, divisions[1:-1])
+    shards = list(map(strip_categories, shards))
```

Comment:
They're not supported by `partd` currently (the intermediate storage used for the shuffle), so we need to remove them beforehand. I don't see any reason why `partd` couldn't be made to support categoricals though.